### ISSUE TITLE
how to use different oxidation states file?

### ISSUE THREAD

**Issue by usccolumbia** [WMD-group/SMACT#48]
[oxidation_states_icsd.txt](https://github.com/WMD-group/SMACT/blob/master/smact/data/oxidation_states_icsd.txt)

e.g. i want to use this icsd oxidation states to do charge neutrality check. how to do it?

**Comment by keeeto** [comment]
@AntObi can you help on this one?

**Comment by dandavies99** [comment]
I don't think this is currently implemented for the charge neutrality check. 

While the ICSD oxidation states [are an element property](https://smact.readthedocs.io/en/latest/smact.html#smact.Element.oxidation_states_icsd), the functions such as `screening.smact_filter` [here](https://smact.readthedocs.io/en/latest/smact.screening.html#smact.screening.smact_filter) always use the default set of oxidation states. From `smact_filter`: 

```python
# Get symbols and electronegativities
    symbols = tuple(e.symbol for e in els)
    electronegs = [e.pauling_eneg for e in els]
    ox_combos = [e.oxidation_states for e in els]
``` 

It would be a fairly trivial addition to add `oxidation_state_set` as an argument to `smact_filter`, which defaults to the standard SMACT `oxidation_states` set, but can be used to switch to the ICSD or Pymatgen set. Perhaps we should implement this.

**Comment by AntObi** [comment]
Closing this issue as the fix has been implemented in version 2.3.2 of smact with an `oxidation_states_set` argument to `smact_filter`.

Usage of the `smact_filter` function is shown below:

```python
# Imports
import smact
from smact.screening import smact_filter

# Let's generate all the possible Fe-O compositions
Fe=smact.Element('Fe')
O=smact.Element('O')

# Create a list of comps using the different oxidation states
default_comps = smact_filter([Fe,O], oxidation_states_set='default')
icsd_comps = smact_filter([Fe,O], oxidation_states_set = 'icsd')
pymatgen_comps = smact_filter([Fe,O], oxidation_states_set='pymatgen')
```

In some cases, the default and icsd oxidation states will result in similar results from `smact_filter`. Typically, the `oxidation_states_set='pymatgen'` will return fewer results.
